pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -508,7 +508,10 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 		}
 		// Push only EDS. This is indexed already - push immediately
 		// (may need a throttle)
-		if len(con.Clusters) > 0 && len(edsUpdatedServices) > 0 {
+		if len(con.Clusters) > 0 {
+			if len(edsUpdatedServices) == 0 {
+				edsUpdatedServices = nil
```

Comment:
Why change this?

Reply:
To get a semantic consistency with PushRequest.merge: empty means all. 
Though it's unlikely for a non-full push having empty edsUpdatedServices.

Reply:
This can be better understood with empty means "not specified". I think the problem comes if "ConfigsChanged" was passed with empty value while full pushing (under some old way), we can't simply regard it as no configs changed and result in no full push.

My personal suggestion: we can keep the special meanings of "empty" but never involves "nil" since it's not a good practice to distinguish "empty" and "nil".